File: Java/PalindromeNumber.java
public class Solution {
    public boolean isPalindrome(int number) {
        if(number < 0){
            return false;
        }
        
        int divisor = 1;
        while(number / divisor >= 10){
            divisor *= 10;
        }
        
        int currLeftDigit  = 0;
        int currRightDigit = 0;
        
        while(number > 0){
            currLeftDigit = number / divisor;
            currRightDigit = number % 10;
            
            number = (number % divisor) / 10;//Remove the right and left most digits.
            divisor = divisor / 100;
            if(currLeftDigit != currRightDigit){
                return false;
            }
        }
        return true;
    }
}

File: C++/PalindromeNumber.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if( x < 0)
            return false;
        int power = 1;
        while(x/power >= 10 )
            power *= 10;
        
        int firstDigit = 0, lastDigit = 0;
        
        while(x && firstDigit == lastDigit){
            lastDigit = x % 10;
            firstDigit = x/power;
            x = (x%power)/10;
            power /=100;
        }
        
        return x == 0 && firstDigit == lastDigit;
    }
};
